StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs

```diff
@@ -175,5 +175,35 @@ internal static bool IsFollowedByWhitespace(this SyntaxToken token)
             triviaList = token.GetNextToken().LeadingTrivia;
             return triviaList.Count > 0 && triviaList.First().IsKind(SyntaxKind.WhitespaceTrivia);
         }
+
+        /// <summary>
+        /// Returns the closest end of line trivia preceding the <paramref name="token"/>.
+        /// This currently only looks immediately before the specified token.
+        /// </summary>
+        /// <param name="token">The token to process.</param>
+        /// <returns>The closest preceding end of line trivia, or <see cref="SyntaxFactory.CarriageReturnLineFeed"/> if none is found.</returns>
+        internal static SyntaxTrivia GetPrecedingEndOfLineTrivia(this SyntaxToken token)
+        {
+            var leadingTrivia = token.LeadingTrivia;
+            for (var i = leadingTrivia.Count - 1; i >= 0; i--)
+            {
+                if (leadingTrivia[i].IsKind(SyntaxKind.EndOfLineTrivia))
+                {
+                    return leadingTrivia[i];
+                }
+            }
+
+            var prevToken = token.GetPreviousToken();
```

Comment:
💡 If there is any way to test this with a preceding `#region` directive, that would be good (preprocessor directives have embedded line endings that are not EndOfLineTrivia). Will leave this for future improvement though.

Reply:
It looks like this rule never triggers when the closing brace is followed by a directive, so I don't see a way to test that case with this rule. But if this method will be used in other code fixers as well, then it seems reasonable to do.